### DISCUSSION TITLE
workaround to deal with `vk::Device::allocateDescriptorSetsUnique: ErrorOutOfPoolMemory`

### DISCUSSION THREAD

**Discussion by keiohta** [haosulab/SAPIEN#78] [Q&A] [answered]
Hi author, I have sometimes encountered an error
```
vk::Device::allocateDescriptorSetsUnique: ErrorOutOfPoolMemory
```
when taking a picture using a camera. In particular, this happens when I call `camera.take_picture()` multiple times (like 10K-100K times) to create a dataset. More concretely,

1. Create an engine
2. For around 1000 URDFs
    1. Create a scene and load an object via URDF
    1. For around 1000 times
        1. Randomize the initial pose of the object
        2. Take some pictures from different angles and save it
    1. Finish this loop **without removing the object and the scene**

Watching used memories for this script using `htop` and `nvidia-smi`, there are enough RAMs for both even when this error terminates the process. In general, do you have any idea when this error likely happens? Or do you know some way to release pool memory for Vulkan Renderer explicitly?

Keeping creating new scenes (and articulations under the scene) can be a problem, but it might not be related to this problem.

I understand this error is not directly related to SAPIEN, but it would be great if you could share your knowledge on this. Thanks in advance for your time.

**Comment by fbxiang** [comment, marked as answer]
You are probably hitting the maximum number of materials and textures. You can increase them by passing larger numbers to the VulkanRenderer constructor.
https://github.com/haosulab/SAPIEN/blob/6bc3f4e2be910199b793f185aea5791d9f193e4c/python/pysapien_content.hpp#L1736-L1737

**Reply by keiohta** [reply]
This perfectly solved the problem I had been struggling with for six months. Thanks a lot!!!
